### PR TITLE
Feat/investigation info tests

### PR DESCRIPTION
**Pull request by y-golde**
**NOTE:** this is the first part of the PR and it doesn't include the big bad and ugly "InvestigatedPersonInfo" for three reasons:
1.  The file is a HUGE mess - including non-repetitive components that should be generified and unrelated/nesting components that should be separated
2.  This PR includes a few important changes to the tests with the additions of new common utils.
3.  This PR is already pretty big and I don't want to overwhelm reviewers
Rest assured because This PR will be followed by a cleanup of InvestigatedPersonInfo and additional tests.

# Changes

### axiosMock.ts
this saves 4 lines of code from any test file that uses MockAdapter from 'axios-mock-adapter' by already creating it
## MutationObserver 
react-hook-form uses mutationObserver - which is normally supplied by the browser (similar to global window or location), added the mutation observer library and an exception at setupTests

### CODE REVIEW

File: client/src/Utils/Testing/InvestigationInfo/MockCommentContextProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (chenFriedman) [inline client/src/Utils/Testing/InvestigationInfo/MockCommentContextProvider.tsx:1]: can remove emptylines

File: client/src/Utils/Testing/axiosMock.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
```
Comment (chenFriedman) [inline client/src/Utils/Testing/axiosMock.ts:2]: 👏

**Review by chenFriedman — APPROVED** [review]

File: client/src/Utils/Testing/InvestigationInfo/MockCommentContextProvider.tsx (hunk 1 above)
Comment (y-golde) [reply client/src/Utils/Testing/InvestigationInfo/MockCommentContextProvider.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1323/commits/0fc6d2a5cb0aeff87014758866de2d4617384629)

**Review by Noam6421 — DISMISSED** [review]

**Review by MagusM — DISMISSED** [review]

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.test.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            changeQuery(wrapper, testQuery);
+
+            expect(wrapper.find(textfieldSelector).props().value).toBe(testQuery);
+        });
+    });
+
+    describe('save button: ', () => {
+        const testQuery = 'test';
+        mockSelectors();
+        const wrapper = mount(
```
[the remark this answers is not in the document]
Comment (MagusM) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.test.tsx:78]: @chenFriedman have you looked at this please ?

[the remark this answers is not in the document]
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.test.tsx:78]: sorry, I forgot, I'll fix it now

[the remark this answers is not in the document]
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.test.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1323/commits/1330c90ea1b4903b50fd252ee5d9e83c9858aa85)

**Review by barnahum8 — APPROVED** [review]

**Review by MagusM — APPROVED** [review]

**Pull request merged**